src/Microsoft.VisualStudio.Services.Agent/Blob/DedupManifestArtifactClientFactory.cs

```diff
@@ -122,8 +129,15 @@ private DedupManifestArtifactClientFactory()
                 continueOnCapturedContext: false);
 
             var telemetry = new BlobStoreClientTelemetry(tracer, dedupStoreHttpClient.BaseAddress);
-            var client = new DedupStoreClientWithDataport(dedupStoreHttpClient, maxParallelism);
-            return (new DedupManifestArtifactClient(telemetry, client, tracer), telemetry);
+            traceOutput($"Hashtype: {this.HashType.Value}");
+
+            if (this.HashType == BuildXL.Cache.ContentStore.Hashing.HashType.Dedup1024K)
+            {
+                dedupStoreHttpClient.RecommendedChunkCountPerCall = 10; // This is to workaround IIS limit - https://learn.microsoft.com/en-us/iis/configuration/system.webserver/security/requestfiltering/requestlimits/
```

Comment:
Not for the PR, but I feel like this setting should be handled more generically - like maybe the dedupclient constructor below.  My concern is that we have to remember to do this for every dedupstorehttpclient.

Reply:
yep, I have created a ADO task specifically for this.